block.go

```diff
@@ -50,11 +51,14 @@ func (b Block) Marshal() []byte {
 
 	n += 4
 
-	for _, id := range b.Transactions {
-		copy(buf[n:n+SizeTransactionID], id[:])
+	var ids []byte
 
-		n += SizeTransactionID
-	}
+	sh := (*reflect.SliceHeader)(unsafe.Pointer(&ids))
```

Comment:
is it worth it using such obscure piece of code for performance?
At least comment should be added, since now its really hard to understand why its done this particular way.

Reply:
Yep, there is a noticeable performance boost as marshaling is done often and blocks often contain thousands of transaction IDs. I'll leave a comment though.